### PR TITLE
Gstreamer plugin running in wayland with surfman 0.2

### LINKED ISSUE #24843: GStreamer plugin upgrade to surfman 0.2.0
The use of surfman in the gst plugin is independent of that in servo, so we can update the plugin before servo itself.

### PR DESCRIPTION
**Pull request by asajeffrey**
Tidying up the gstreamer plugin. The plugin now:

* uses surfman 0.2
* runs in wayland (but can't render WebGL content yet)
* gets its GL configuration from gstreamer
* uses GLsync if needed

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #24843 
- [X] These changes do not require tests because

### CODE REVIEW

**Comment by asajeffrey** [comment]
Oops, should have squashed before submitting the PR!

File: ports/gstplugin/Cargo.toml [hunk 1]
```diff
@@ -31,9 +31,9 @@ libservo = {path = "../../components/servo"}
 servo-media = {git = "https://github.com/servo/media"}
 sparkle = "0.1"
 # NOTE: the sm-angle-default feature only enables angle on windows, not other platforms!
```
Comment (jdm) [inline ports/gstplugin/Cargo.toml]: Remove this note now that it no longer applies?

File: ports/gstplugin/servowebsrc.rs [hunk 2]
```diff
@@ -663,10 +729,20 @@ impl BaseSrcImpl for ServoWebSrc {
...
         unsafe { gst_gl_context_thread_add(gl_memory.mem.context, Some(fill_on_gl_thread), data) };
         task.result?;
 
+        // Put down a GL sync point if needed
+        let gst_buffer = buffer.to_glib_none();
+        let sync_meta_type = unsafe { gst_gl_sync_meta_api_get_type() };
+        let sync_meta =
+            unsafe { gst_buffer_get_meta(gst_buffer.0, sync_meta_type) } as *mut GstGLSyncMeta;
+        if !sync_meta.is_null() {
+            unsafe { gst_gl_sync_meta_set_sync_point(sync_meta, gl_memory.mem.context) };
```
Comment (jdm) [inline ports/gstplugin/servowebsrc.rs]: We should be able to use:
* https://slomo.pages.freedesktop.org/rustdocs/gstreamer/gstreamer/buffer/struct.BufferRef.html#method.get_meta
* https://slomo.pages.freedesktop.org/rustdocs/gstreamer/gstreamer_gl/struct.GLSyncMeta.html
* https://slomo.pages.freedesktop.org/rustdocs/gstreamer/gstreamer_gl/struct.GLSyncMeta.html#method.set_sync_point

**Review by jdm — CHANGES_REQUESTED** [review]

File: ports/gstplugin/Cargo.toml (hunk 1 above)
Comment (asajeffrey) [reply ports/gstplugin/Cargo.toml]: Done.

File: ports/gstplugin/servowebsrc.rs (hunk 2 above)
Comment (asajeffrey) [reply ports/gstplugin/servowebsrc.rs]: Done.

**Review by asajeffrey — COMMENTED** [review]
Fixed.

**Comment by jdm** [comment]
Windows plugin build is broken:
```
error[E0412]: cannot find type `Device` in this scope
   --> ports\gstplugin\servowebsrc.rs:103:17
    |
103 | type Context = <Device as DeviceAPI>::Context;
    |                 ^^^^^^ not found in this scope
    |
help: possible candidates are found in other modules, you can import them into scope
    |
5   | use gstreamer::Device;
    |
5   | use surfman::Device;
    |
5   | use surfman::device::Device;
    |
5   | use surfman::platform::default::device::Device;
    |
      and 1 other candidate

error[E0412]: cannot find type `Device` in this scope
   --> ports\gstplugin\servowebsrc.rs:104:20
    |
104 | type Connection = <Device as DeviceAPI>::Connection;
    |                    ^^^^^^ not found in this scope
    |
help: possible candidates are found in other modules, you can import them into scope
    |
5   | use gstreamer::Device;
    |
5   | use surfman::Device;
    |
5   | use surfman::device::Device;
    |
5   | use surfman::platform::default::device::Device;
    |
      and 1 other candidate

error[E0412]: cannot find type `Device` in this scope
   --> ports\gstplugin\servowebsrc.rs:105:23
    |
105 | type NativeContext = <Device as DeviceAPI>::NativeContext;
    |                       ^^^^^^ not found in this scope
    |
help: possible candidates are found in other modules, you can import them into scope
    |
5   | use gstreamer::Device;
    |
5   | use surfman::Device;
    |
5   | use surfman::device::Device;
    |
5   | use surfman::platform::default::device::Device;
    |
      and 1 other candidate

error[E0412]: cannot find type `Device` in this scope
   --> ports\gstplugin\servowebsrc.rs:138:13
    |
138 |     device: Device,
    |             ^^^^^^ not found in this scope
    |
help: possible candidates are found in other modules, you can import them into scope
    |
5   | use gstreamer::Device;
    |
5   | use surfman::Device;
    |
5   | use surfman::device::Device;
    |
5   | use surfman::platform::default::device::Device;
    |
      and 1 other candidate

error[E0412]: cannot find type `Device` in this scope
   --> ports\gstplugin\servowebsrc.rs:140:27
    |
140 |     swap_chain: SwapChain<Device>,
    |                           ^^^^^^ not found in this scope
    |
help: possible candidates are found in other modules, you can import them into scope
    |
5   | use gstreamer::Device;
    |
5   | use surfman::Device;
    |
5   | use surfman::device::Device;
    |
5   | use surfman::platform::default::device::Device;
    |
      and 1 other candidate

error[E0412]: cannot find type `Device` in this scope
   --> ports\gstplugin\servowebsrc.rs:168:35
    |
168 |     GetSwapChain(Sender<SwapChain<Device>>),
    |                                   ^^^^^^ not found in this scope
    |
help: possible candidates are found in other modules, you can import them into scope
    |
5   | use gstreamer::Device;
    |
5   | use surfman::Device;
    |
5   | use surfman::device::Device;
    |
5   | use surfman::platform::default::device::Device;
    |
      and 1 other candidate

error[E0412]: cannot find type `Device` in this scope
   --> ports\gstplugin\servowebsrc.rs:182:27
    |
182 |     swap_chain: SwapChain<Device>,
    |                           ^^^^^^ not found in this scope
    |
help: possible candidates are found in other modules, you can import them into scope
    |
5   | use gstreamer::Device;
    |
5   | use surfman::Device;
    |
5   | use surfman::device::Device;
    |
5   | use surfman::platform::default::device::Device;
    |
      and 1 other candidate

error[E0412]: cannot find type `Device` in this scope
   --> ports\gstplugin\servowebsrc.rs:188:13
    |
188 |     device: Device,
    |             ^^^^^^ not found in this scope
    |
help: possible candidates are found in other modules, you can import them into scope
    |
5   | use gstreamer::Device;
    |
5   | use surfman::Device;
    |
5   | use surfman::device::Device;
    |
5   | use surfman::platform::default::device::Device;
    |
      and 1 other candidate

error[E0412]: cannot find type `Device` in this scope
   --> ports\gstplugin\servowebsrc.rs:292:27
    |
292 |     swap_chain: SwapChain<Device>,
    |                           ^^^^^^ not found in this scope
    |
help: possible candidates are found in other modules, you can import them into scope
    |
5   | use gstreamer::Device;
    |
5   | use surfman::Device;
    |
5   | use surfman::device::Device;
    |
5   | use surfman::platform::default::device::Device;
    |
      and 1 other candidate

error[E0282]: type annotations needed for `std::sync::Mutex<std::option::Option<T>>`
   --> ports\gstplugin\servowebsrc.rs:497:37
    |
497 |         let connection = Mutex::new(None);
    |             ----------              ^^^^ cannot infer type for `T`
    |             |
    |             consider giving `connection` the explicit type `std::sync::Mutex<std::option::Option<T>>`, where the type parameter `T` is specified

error: unused import: `surfman_chains_api::SwapChainAPI`
  --> ports\gstplugin\servowebsrc.rs:95:5
   |
95 | use surfman_chains_api::SwapChainAPI;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D unused-imports` implied by `-D warnings`

error: aborting due to 11 previous errors
```

**Comment by asajeffrey** [comment]
Filed #25353, windows support will have to wait for me having a windows machine to test on.

**Pull request merged**